include/zephyr/net/ethernet.h

```diff
@@ -26,8 +26,12 @@
 #include <zephyr/net/ethernet_vlan.h>
 #include <zephyr/net/ptp_time.h>
 
+#if defined(CONFIG_NET_DSA)
```

Comment:
Usually it is not needed to add ifdef for including public header files. This helps to avoid clutter *.c code with ifdefs.

Reply:
Get it. Let me remove it.

Reply:
Although I removed ifdef for CONFIG_NET_DSA here, the twister issues showed that dsa_core.h including ``zephyr/drivers/pinctrl.h`` also must require ``pinctrl_soc.h``.
I actually intended to make pinctrl as optional in dsa, should I wrapper pinctrl header/code with ifdef?
Or there may be better idea?

Thank you very much.